collections/src/main/scala/strawman/collection/immutable/LazyList.scala

```diff
@@ -535,6 +548,9 @@ object LazyList extends LazyListFactory[LazyList] {
 
 @deprecated("Use LazyList (which has a lazy head and tail) instead of Stream (which has a lazy tail only)", "2.13.0")
 sealed abstract class Stream[+A] extends LinearSeq[A] with LazyListOps[A, Stream, Stream[A]] {
```

Comment:
For `Stream` I would just copy the old implementation.

Reply:
By 'old' you mean to say Scala 2.12 ? If so, I will have a look. I have made changes to this one after your last review to use `StringBuilder` so it should be in terms of performance and have added unit tests as well.

Reply:
Just checked scala 2.12 `Stream#toString` [implementation](https://github.com/scala/scala/blob/80dc0d9dd6ab783c0c54afac7eadf88c93bd02a7/src/library/scala/collection/immutable/Stream.scala#L669). I have forgotten to take care of cyclic streams. BTW, is it realistically possible for library user to create cyclic `LazyList` ?

Reply:
> is it realistically possible for library user to create cyclic LazyList

I hope so — if not, it seems like a bug to me (and a regression since `Stream`)

Reply:
Yes it is: https://scastie.scala-lang.org/jvN81N4QQ5CDuETqcbF0Jg